go/tasks/plugins/array/awsbatch/job_definition.go

```diff
@@ -16,6 +16,9 @@ import (
 	"github.com/flyteorg/flyteplugins/go/tasks/plugins/array/awsbatch/definition"
 )
 
+const defaultComputeEngine = "EC2"
+const platformCapabilities = "platformCapabilities"
```

Comment:
nit: let's try to avoid [shadowing variables](https://yourbasic.org/golang/gotcha-shadowing-variables/) in general. It leads to confused unhappy maintainers...
Maybe name it something like:
```suggestion
const platformCapabilitiesConfigKey = "platformCapabilities"
```